proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -304,15 +440,92 @@ elimation has completed.
 
 ### Validations in Sema
 
-TODO
+#### All the values should be legal.
+
+Most values like ShaderVisibility/ParameterType are covered by syntactical checks in Sema.
+Only list special rule here.
+
+##### DescriptorRangeFlags
+
+###### Sampler
+- None
+- DescriptorsVolatile
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
```

Comment:
I can't figure out what any of these mean.  Can you make it super clear and explicit what the is being validated here? The only references to DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS I can find are in this section on validation.  What does it mean? Is this meant to a supported flag somewhere?

What is "DescriptorsVolatile"?

Reply:
Here's doc for [DescriptorsVolatile](https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signature-version-1-1#descriptors_volatile) .
Cannot find doc about DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS as well :(
But from the last part of the [DescriptorsVolatile](https://learn.microsoft.com/en-us/windows/win32/direct3d12/root-signature-version-1-1#descriptors_volatile), it might be used when depend on safe out of bounds memory access behavior when accessing descriptors, but don't want to set DescriptorsVolatile.